haystack/document_stores/sql.py

```diff
@@ -164,6 +150,8 @@ def __init__(
         :param check_same_thread: Set to False to mitigate multithreading issues in older SQLite versions (see https://docs.sqlalchemy.org/en/14/dialects/sqlite.html?highlight=check_same_thread#threading-pooling-behavior)
         :param isolation_level: see SQLAlchemy's `isolation_level` parameter for `create_engine()` (https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.isolation_level)
         """
+        sqlalchemy_import.check()
```

Comment:
No strong opinions, but I think a comment might be useful here
```suggestion
        # ensure the required dependencies were actually imported
        sqlalchemy_import.check()
```